File: sdk/node/examples/agents_md_demo.ts
/**
 * Example: AGENTS.md Auto-Loading
 *
 * Demonstrates automatic loading of AGENTS.md from workspace root.
 * Similar to Claude Code's CLAUDE.md mechanism.
 */

import { Agent } from '..';
import * as fs from 'fs';
import * as path from 'path';
import * as os from 'os';

async function main() {
  console.log('🚀 AGENTS.md Auto-Loading Example\n');

  // Create a temporary workspace with AGENTS.md
  const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'agents-md-test-'));
  console.log(`📁 Workspace: ${tmpDir}\n`);

  // Write AGENTS.md with project-specific instructions
  const agentsMdContent = `# Project Instructions

This is a TypeScript project using Node.js and Express.

## Code Style
- Use TypeScript strict mode
- Prefer async/await over callbacks
- Use ESLint and Prettier
- Write tests with Jest

## Architecture
- Follow MVC pattern
- Use dependency injection
- Keep controllers thin
- Business logic in services

## Testing
- Unit tests for all services
- Integration tests for API endpoints
- Minimum 80% code coverage

## Security
- Validate all user input
- Use parameterized queries
- Never log sensitive data
- Follow OWASP Top 10 guidelines
`;

  fs.writeFileSync(path.join(tmpDir, 'AGENTS.md'), agentsMdContent);
  console.log('✅ Created AGENTS.md in workspace\n');

  // Create agent and session
  const agent = await Agent.create('agent.hcl');
  const session = agent.session(tmpDir, {
    builtinSkills: true,
  });

  console.log('📝 Sending prompt to agent...\n');

  // Send a prompt - the agent should follow AGENTS.md instructions
  const result = await session.send(
    'Create a new user registration endpoint with validation and tests'
  );

  console.log('✅ Agent response:\n');
  console.log(result.text);
  console.log(`\n📊 Stats: ${result.toolCallsCount} tools, ${result.totalTokens} tokens`);

  // Cleanup
  fs.rmSync(tmpDir, { recursive: true, force: true });
  console.log('\n🧹 Cleaned up temporary workspace');
}

main().catch(console.error);

File: sdk/python/examples/agents_md_demo.py
"""
Example: AGENTS.md Auto-Loading

Demonstrates automatic loading of AGENTS.md from workspace root.
Similar to Claude Code's CLAUDE.md mechanism.
"""

from a3s_code import Agent
import tempfile
import os

def main():
    print("🚀 AGENTS.md Auto-Loading Example\n")

    # Create a temporary workspace with AGENTS.md
    with tempfile.TemporaryDirectory(prefix="agents-md-test-") as tmpdir:
        print(f"📁 Workspace: {tmpdir}\n")

        # Write AGENTS.md with project-specific instructions
        agents_md_content = """# Project Instructions

This is a Python project using FastAPI and SQLAlchemy.

## Code Style
- Use type hints everywhere
- Follow PEP 8
- Use Black for formatting
- Use mypy for type checking

## Architecture
- Follow Clean Architecture
- Use dependency injection
- Keep routes thin
- Business logic in services

## Testing
- Use pytest for all tests
- Unit tests for all services
- Integration tests for API endpoints
- Minimum 80% code coverage

## Security
- Validate all user input with Pydantic
- Use parameterized queries
- Never log sensitive data
- Follow OWASP Top 10 guidelines
"""

        agents_md_path = os.path.join(tmpdir, "AGENTS.md")
        with open(agents_md_path, "w") as f:
            f.write(agents_md_content)

        print("✅ Created AGENTS.md in workspace\n")

        # Create agent and session
        agent = Agent.create("agent.hcl")
        session = agent.session(tmpdir, builtin_skills=True)

        print("📝 Sending prompt to agent...\n")

        # Send a prompt - the agent should follow AGENTS.md instructions
        result = session.send(
            "Create a new user registration endpoint with validation and tests"
        )

        print("✅ Agent response:\n")
        print(result.text)
        print(f"\n📊 Stats: {result.tool_calls_count} tools, {result.total_tokens} tokens")

        print("\n🧹 Workspace will be cleaned up automatically")

if __name__ == "__main__":
    main()
